### PR TITLE
Gradle init script improvements

### PR DESCRIPTION
**Pull request by bigdaz**
Various refactorings and improvements to the init-scripts used to instrument Gradle:
- Combine 3 separate init-scripts that were configuring GE into a single script.
- Improve error message when build-validation was run on a non-GE project (without `-e`).
- Avoid duplicate plugin application when running on a GE-enabled project with `-e`.

As well as improving init-script behaviour, these increase the similarity between init-scripts in this project and the TC Build Scan plugin.

### CODE REVIEW

**Comment by bigdaz** [comment]
@etiennestuder I've made the init-scripts as close as I can to the "gold standard" without too many contortions. There are some fundamental differences that make it hard to get closer (ie the build-validation scripts _always_ override the GE server config, but the TC scripts only do this when adding GE to a non-GE project).

I guess a next step could be to try to enable build-validation for projects using Gradle < 5. I haven't tested this because I don't currently have a GE server that allows anonymous publishing, and Gradle 4 doesn't support authentication.

File: components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle [hunk 1]
```diff
@@ -1,64 +1,159 @@
...
+def isTopLevelBuild = !gradle.parent
+if (!isTopLevelBuild) {
+    return
+}
+
+def enableGradleEnterprise = Boolean.getBoolean("com.gradle.enterprise.build_validation.enableGradleEnterprise")
+
+clearWarnings()
+
+// define a buildScanPublished listener that captures the build scan URL and sends it in a message to the server
```
Comment (jthurne) [inline components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle:56]: -1: Looks like an inaccurate comment.

Comment (bigdaz) [reply components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle:56]: Good point. This was copied from the "Gold standard" script and should be modified as appropriate

**Review by jthurne — APPROVED** [review]

**Comment by etiennestuder** [comment]
I'll take a look over the week-end while in the air.

**Comment by bigdaz** [comment]
@etiennestuder Are you still interested in these changes? If so I can rebase and resolve and conflicts.

**Comment by etiennestuder** [comment]
@bigdaz I'm very mnuch still interested. Just did not get to it, yet. Thanks for resolving the coflicts (I belive it's only chnaged version of GE and CCUD plugin).

**Comment by bigdaz** [comment]
@etiennestuder PR has been rebased on `main` with conflicts resolved.
Note that I remember seeing some further improvements that I'd like to make to the init-script, but these would require changes to the TC plugin as well to keep things in sync.

**Comment by erichaagdev** [comment]
Closing this PR in favor of #264

**Pull request closed without merging**